core/src/test/kotlin/integration_tests/JSONObjectPatternExampleIntegrationTest.kt

```diff
@@ -0,0 +1,483 @@
+package integration_tests
+
+import io.specmatic.conversions.OpenApiSpecification
+import io.specmatic.core.HttpRequest
+import io.specmatic.core.pattern.parsedJSONObject
+import io.specmatic.stub.HttpStub
+import io.specmatic.test.TestExecutor
+import io.specmatic.core.HttpResponse
+import io.specmatic.core.value.Value
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Nested
+
+class JSONObjectPatternExampleIntegrationTest {
+
+    @Nested
+    inner class RequestExampleTests {
+
+        @Test
+        fun `should use JSON object example in request when running contract tests`() {
+            val contract = OpenApiSpecification.fromYAML(
+                """
+                openapi: 3.0.0
+                info:
+                  title: User API
+                  version: "1.0"
+                paths:
+                  /users:
+                    post:
+                      summary: Create user
+                      requestBody:
+                        content:
+                          application/json:
+                            schema:
+                              type: object
+                              required:
+                                - name
+                                - email
+                              properties:
+                                name:
+                                  type: string
+                                email:
+                                  type: string
+                                age:
+                                  type: integer
+                            examples:
```

Comment:
Currently the test has named examples. Instead, the examples should look like this:

```json
type: object
required:
  ...
properties:
  ...
examples:
    CREATE_USER:
      value:
        name: "John Doe"
        email: "john@example.com"
        age: 30
```

- The same in response. Update the other newly added tests in this file.
- Update `OpenApiSpecification.toJsonObjectPattern` to store the the example in JSONObjectPattern.example.
- Then update JSONObjectPattern to ensure that this test and other newly added tests using OpenAPI specifications pass.